lib/Crypto/cryptodaemonconnection_p.h

```diff
@@ -8,35 +8,52 @@
 #ifndef LIBSAILFISHCRYPTO_CRYPTODAEMONCONNECTION_P_H
 #define LIBSAILFISHCRYPTO_CRYPTODAEMONCONNECTION_P_H
 
-#include "Crypto/cryptodaemonconnection.h"
+// WARNING!
+//
+// This is private API, used for internal implementation only!
+// No BC/SC guarantees are made for the methods in this file!
+
+#include "Crypto/cryptoglobal.h"
 
 #include <QtDBus/QDBusConnection>
 #include <QtDBus/QDBusInterface>
 
 #include <QtCore/QObject>
-#include <QtCore/QPointer>
+#include <QtCore/QString>
+#include <QtCore/QAtomicInt>
+#include <QtCore/QLoggingCategory>
+
+Q_DECLARE_LOGGING_CATEGORY(lcSailfishCryptoDaemonConnection)
 
 namespace Sailfish {
 
 namespace Crypto {
 
-class CryptoDaemonConnection;
-class CryptoDaemonConnectionPrivate : public QObject
+class CryptoDaemonConnectionPrivate;
+class SAILFISH_CRYPTO_API CryptoDaemonConnection : public QObject
 {
     Q_OBJECT
 
 public:
-    CryptoDaemonConnectionPrivate(CryptoDaemonConnection *parent = Q_NULLPTR);
-    QDBusConnection *connection() { return &m_connection; }
+    static CryptoDaemonConnection *instance();
+    static void releaseInstance();
+
     bool connect();
+    QDBusConnection *connection();
+    QDBusInterface *createInterface(const QString &objectPath,
+                                    const QString &interface,
+                                    QObject *parent = Q_NULLPTR);
+
+    static void registerDBusTypes();
 
-public Q_SLOTS:
+signals:
```

Comment:
Some incosistency: you use `Q_SIGNALS` most of time but here is `signals:`.

Reply:
Thanks, fixed